### PR TITLE
Update sdk to 3.1.1

### LINKED ISSUE #32: Add support for Classic L1-to-L2 messages
[v3.1.0](https://www.npmjs.com/package/@arbitrum/sdk/v/3.1.0) of the SDK adds support, and we should integrate it in this repo

### PR DESCRIPTION
**Pull request by chrstph-dvx**
- [X] Bump Arbitrum SDK to 3.1.1
- [X] Fix issue when clicking on redeem with wallet already connected woul show an error message (https://github.com/OffchainLabs/retryable-tx-panel/pull/36)

Fix #32

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -414,10 +455,10 @@ function App() {
     }
   }, [navigate, oldTxHash]);
 
-  const { data: signer = null } = useSigner();
+  const { chain } = useNetwork();
+  const { data: signer = null } = useSigner({ chainId: chain?.id });
```
Comment (chrstph-dvx) [inline src/App.tsx:459]: @DZGoldman This fixes the issue with "underlying network changed"

File: src/Redeem.tsx [hunk 2]
```diff
@@ -30,20 +30,19 @@ function Redeem({
             return;
           }
 
-          const _l1ToL2Message: {
-            sender: string;
-            messageNumber: BigNumber;
-            l1BaseFee: BigNumber;
-            messageData: RetryableMessageParams;
-          } = l1ToL2Message.l1ToL2Message;
+          // @ts-ignore it works
```
Comment (DZGoldman) [inline src/Redeem.tsx:33]: mood

File: src/App.tsx [hunk 3]
```diff
@@ -244,22 +253,38 @@ const getL1ToL2MessagesAndDepositMessages = async (
         );
     }
     const l2Provider = new StaticJsonRpcProvider(l2RpcURL);
-    const l1ToL2MessagesWithNetwork: L1ToL2MessageReaderWithNetwork[] = (
-      await l1TxnReceipt.getL1ToL2Messages(l2Provider)
-    ).map((l1ToL2Message) => {
-      return Object.assign(l1ToL2Message, { l2Network });
-    });
+    const isClassic = await l1TxnReceipt.isClassic(l2Provider);
```
Comment (DZGoldman) [inline src/App.tsx:256]: we shouldn't ever be seeing classic l1 to l2 messages anymore, so it seems fine to disregard IMO (also fine to leave as is)

**Review by DZGoldman — APPROVED** [review]
lgtm

**Review by gzeoneth — APPROVED** [review]
LGTM

**Pull request merged**